<details><summary>Failed to submit graph for 9ba4239fa996d8e2cc9feb83d3ff43b574efec5f
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmplsobh8me/pytorch-feedstock.yaml --custom-metadata-file /tmp/tmplsobh8me/custom_metadata.yaml --release-name pytorch-feedstock-95-9ba4239fa996d8e2cc9feb83d3ff43b574efec5f pytorch-feedstock/recipe', with error: 'INFO:root:Found feedstock pytorch-feedstock at /tmp/tmplsobh8me/pytorch-feedstock/recipe
INFO:root:Build/Test graph from pytorch-feedstock
Created by sbx_user1051 on 2026-05-02 21:04:14
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmplsobh8me/pytorch-feedstock.yaml --custom-metadata-file /tmp/tmplsobh8me/custom_metadata.yaml --release-name pytorch-feedstock-95-9ba4239fa996d8e2cc9feb83d3ff43b574efec5f pytorch-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmplsobh8me/pytorch-feedstock
INFO:root:Creating graph from feedstock at: pytorch-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.12 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 5 unchanging vars:(464), changing vars (5): ['blas_impl', 'cuda_compiler_version', 'gpu_variant', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_2mcmwzoz.
WARNING: Adding .* to spec 'cuda-version 13.1' to ensure satisfiability.  Please consider putting {{{{ var_name }}}}.* or some relational operator (>/</>=/<=) on this spec in meta.yaml, or if req is also a build req, using {{{{ pin_compatible() }}}} jinja2 function instead.  See https://conda.io/docs/user-guide/tasks/build-packages/variants.html#pinning-at-the-variant-level
INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 416, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 643, in from_feedstock
    build_configs, test_configs = expand_build_matrix(
                                  ~~~~~~~~~~~~~~~~~~~^
        feedstock.recipe_dir_path, src_channels, parsed_user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 82, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels, user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 566, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=user_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 68, in render
    return render_metadata_tuples(
        metadata_tuples,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1097, in render_metadata_tuples
    for od, om in meta.get_output_metadata_set(
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        permit_undefined_jinja=not finalize,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2781, in get_output_metadata_set
    conda_packages = finalize_outputs_pass(
        ref_metadata,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1077, in finalize_outputs_pass
    fm = finalize_metadata(
        om,
        parent_metadata=parent_metadata,
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 722, in finalize_metadata
    build_unsat, host_unsat = add_upstream_pins(
                              ~~~~~~~~~~~~~~~~~^
        m, permit_unsatisfiable_variants, exclude_pattern, extra_specs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 507, in add_upstream_pins
    host_deps, host_unsat, extra_run_specs_from_host = _read_upstream_pin_files(
                                                       ~~~~~~~~~~~~~~~~~~~~~~~~^
        m, "host", permit_unsatisfiable_variants, exclude_pattern, extra_specs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 463, in _read_upstream_pin_files
    extra_run_specs = get_upstream_pins(m, precs, env)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 433, in get_upstream_pins
    loc, dist = execute_download_actions(
                ~~~~~~~~~~~~~~~~~~~~~~~~^
        m,
        ^^
    ...<2 lines>...
        package_subset=[prec],
        ^^^^^^^^^^^^^^^^^^^^^^
    )[prec]
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 396, in execute_download_actions
    pfe.execute()
    ~~~~~~~~~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda/core/package_cache_data.py", line 925, in execute
    raise CondaMultiError(not_cancelled)
conda.CondaMultiError: Failed to write to /tmp/conda_pkgs_2mcmwzoz/magma-2.10.0-h5c33853_0.conda.partial
  errno: 28
Failed to write to /tmp/conda_pkgs_2mcmwzoz/magma-2.10.0-h5c33853_0.conda.partial
  errno: 28

'
aws lambda request id: a5d5729c-d053-5e55-aa1a-10ecb8a9bae3
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%60a5d5729c-d053-5e55-aa1a-10ecb8a9bae3%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -70,10 +70,6 @@ source:
 
 build:
   number: {{ build }}
-  skip: true  # [not linux]
-  skip: true  # [gpu_variant == "cpu"]
-  skip: true  # [blas_impl != "openblas"]
-  skip: true  # [py != 313]
   string: gpu_cuda{{ cuda_compiler_version | replace('.', '') }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [gpu_variant == "cuda"]
   string: gpu_mps_h{{PKG_HASH}}_{{ PKG_BUILDNUM }}                                                   # [gpu_variant == "metal"]
   string: cpu_{{ blas_impl }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                     # [gpu_variant == "cpu"]

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -70,6 +70,10 @@ source:
 
 build:
   number: {{ build }}
+  skip: true  # [not linux]
+  skip: true  # [gpu_variant == "cpu"]
+  skip: true  # [blas_impl != "openblas"]
+  skip: true  # [py != 313]
   string: gpu_cuda{{ cuda_compiler_version | replace('.', '') }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [gpu_variant == "cuda"]
   string: gpu_mps_h{{PKG_HASH}}_{{ PKG_BUILDNUM }}                                                   # [gpu_variant == "metal"]
   string: cpu_{{ blas_impl }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                     # [gpu_variant == "cpu"]

--- a/recipe/patches/0024-Allow-fbgemm_genai-to-be-assigned-to-a-Ninja-job-pool.patch
+++ b/recipe/patches/0024-Allow-fbgemm_genai-to-be-assigned-to-a-Ninja-job-pool.patch
@@ -1,24 +1,24 @@
-From c4433218ad09e4d0c794046b42e30e71c62a60c3 Mon Sep 17 00:00:00 2001
+From 67170d4e398c75e9ce912d6a078ac4a7fe70676b Mon Sep 17 00:00:00 2001
 From: Jamie Robertson <jamierobertsongames@gmail.com>
 Date: Thu, 23 Apr 2026 21:37:14 +0100
 Subject: [PATCH] Allow fbgemm_genai to be assigned to a Ninja job pool
 
-The CUTLASS-based f4f4bf16 / mx8mx8bf16 kernels in fbgemm_genai peak at multiple GB of RSS per nvcc invocation during cicc template instantiation and routinely OOM memory-constrained build hosts (~32GB runners). Globally throttling MAX_JOBS to work around this penalises the rest of the build, which is cheap by comparison.
+The CUTLASS-based f4f4bf16 / mx8mx8bf16 kernels in the MSLK/fbgemm_genai build target peak at multiple GB of RSS per nvcc invocation during cicc template instantiation and routinely OOM memory-constrained build hosts (~32GB runners). Globally throttling MAX_JOBS to work around this penalises the rest of the build, which is cheap by comparison.
 
-Expose an optional FBGEMM_GENAI_JOB_POOL CMake variable that, when set, tags the fbgemm_genai target with JOB_POOL_COMPILE so Ninja can serialize only these compiles via CMAKE_JOB_POOLS while the rest of the build continues at normal parallelism. When the variable is not set, upstream build behaviour is unchanged.
+Expose an optional FBGEMM_GENAI_JOB_POOL CMake variable that, when set, tags the target with JOB_POOL_COMPILE so Ninja can serialize only these compiles via CMAKE_JOB_POOLS while the rest of the build continues at normal parallelism. When the variable is not set, upstream build behaviour is unchanged.
 ---
  aten/src/ATen/CMakeLists.txt | 14 ++++++++++++++
  1 file changed, 14 insertions(+)
 
 diff --git a/aten/src/ATen/CMakeLists.txt b/aten/src/ATen/CMakeLists.txt
-index a3e841ca22a..bbef9285f07 100644
+index 5cc22852618..923af45327a 100644
 --- a/aten/src/ATen/CMakeLists.txt
 +++ b/aten/src/ATen/CMakeLists.txt
-@@ -290,6 +290,20 @@ IF(USE_FBGEMM_GENAI)
+@@ -392,6 +392,20 @@ IF(USE_MSLK)
  
-     set_target_properties(fbgemm_genai PROPERTIES POSITION_INDEPENDENT_CODE ON)
+     set_target_properties(mslk PROPERTIES POSITION_INDEPENDENT_CODE ON)
  
-+    # Optional: assign the fbgemm_genai target to a dedicated Ninja job pool
++    # Optional: assign the mslk target to a dedicated Ninja job pool
 +    # so downstream packagers can serialize the CUTLASS-heavy f4/mx8 compiles.
 +    # These kernels can peak at multiple GB of RSS per nvcc invocation and OOM
 +    # memory-constrained build hosts. The pool name supplied here must also be
@@ -28,13 +28,13 @@ index a3e841ca22a..bbef9285f07 100644
 +    # The USE_ prefix lets PyTorch's setup.py auto-forward the variable from
 +    # the environment. When unset, upstream build behaviour is unchanged.
 +    if(USE_FBGEMM_GENAI_JOB_POOL)
-+      set_target_properties(fbgemm_genai PROPERTIES
++      set_target_properties(mslk PROPERTIES
 +        JOB_POOL_COMPILE "${USE_FBGEMM_GENAI_JOB_POOL}")
 +    endif()
 +
-     set(fbgemm_genai_cuh
-       "${FBGEMM_GENAI_SRCS}/cutlass_extensions/mx8mx8bf16_grouped/"
-       "${FBGEMM_GENAI_SRCS}/cutlass_extensions/f4f4bf16_grouped/"
+     set(mslk_cuh
+       "${MSLK_SRCS}/cutlass/mx8mx8bf16_grouped/"
+       "${MSLK_SRCS}/cutlass/f4f4bf16_grouped/"
 -- 
 2.50.1 (Apple Git-155)
 

--- a/recipe/patches/0024-Allow-fbgemm_genai-to-be-assigned-to-a-Ninja-job-pool.patch
+++ b/recipe/patches/0024-Allow-fbgemm_genai-to-be-assigned-to-a-Ninja-job-pool.patch
@@ -1,19 +1,11 @@
-From a4312a33c2f425ed738a63bf4cbf289513f4740c Mon Sep 17 00:00:00 2001
+From c4433218ad09e4d0c794046b42e30e71c62a60c3 Mon Sep 17 00:00:00 2001
 From: Jamie Robertson <jamierobertsongames@gmail.com>
 Date: Thu, 23 Apr 2026 21:37:14 +0100
 Subject: [PATCH] Allow fbgemm_genai to be assigned to a Ninja job pool
 
-The CUTLASS-based f4f4bf16 / mx8mx8bf16 kernels in fbgemm_genai peak at
-multiple GB of RSS per nvcc invocation during cicc template
-instantiation and routinely OOM memory-constrained build hosts
-(~32GB runners). Globally throttling MAX_JOBS to work around this
-penalises the rest of the build, which is cheap by comparison.
+The CUTLASS-based f4f4bf16 / mx8mx8bf16 kernels in fbgemm_genai peak at multiple GB of RSS per nvcc invocation during cicc template instantiation and routinely OOM memory-constrained build hosts (~32GB runners). Globally throttling MAX_JOBS to work around this penalises the rest of the build, which is cheap by comparison.
 
-Expose an optional FBGEMM_GENAI_JOB_POOL CMake variable that, when
-set, tags the fbgemm_genai target with JOB_POOL_COMPILE so Ninja can
-serialize only these compiles via CMAKE_JOB_POOLS while the rest of
-the build continues at normal parallelism. When the variable is not
-set, upstream build behaviour is unchanged.
+Expose an optional FBGEMM_GENAI_JOB_POOL CMake variable that, when set, tags the fbgemm_genai target with JOB_POOL_COMPILE so Ninja can serialize only these compiles via CMAKE_JOB_POOLS while the rest of the build continues at normal parallelism. When the variable is not set, upstream build behaviour is unchanged.
 ---
  aten/src/ATen/CMakeLists.txt | 14 ++++++++++++++
  1 file changed, 14 insertions(+)

